tools/automation/cli_linter/rules/help_rules.py

```diff
@@ -44,3 +47,109 @@ def faulty_help_example_rule(linter, help_entry):
     if violations:
         raise RuleError('The following example entry indices do not include the command: {}'.format(
             ' | '.join(violations)))
+
+@help_file_entry_rule
+def faulty_help_example_parameters_rule(linter, help_entry):
+    parser = linter.command_parser
+    violations = []
+
+    for example in linter.get_help_entry_examples(help_entry):
+        max_profile = example.get('max_profile')
+        if max_profile and max_profile != 'latest':
+            logger.warning("\n\tSKIPPING example: {}\n\tas its max profile is {}, instead of latest.".format(example['text'], example['max_profile']))
+            continue
+
+        example_text = example.get('text','')
+        commands = _extract_commands_from_example(example_text)
+        while commands:
+            command = commands.pop()
+            violation, nested_commands = _lint_example_command(command, parser)
+
+            commands.extend(nested_commands)  # append commands that are the source of any arguments
+            if violation:
+                violations.append(violation)
+
+    if violations:
+        num_err = len(violations)
+        violation_str = "\n\n".join(violations)
+        violation_msg = "\n\tThere is a violation:\n{}.".format(violation_str) if num_err == 1 else \
+            "\n\tThere are {} violations:\n{}".format(num_err, violation_str)
+        raise RuleError(violation_msg + "\n\n")
```

Comment:
It looks like this:
```
Initializing linter with command table and help files...

	SKIPPING example: az storage account create -n MyStorageAccount -g MyResourceGroup -l westus --account-type Standard_LRS
	as its max profile is 2017-03-09-profile, instead of latest.
-  FAIL: faulty_help_example_parameters_rule
    Help-Entry: `acr build-task create` - 
	There is a violation:
	"az acr build-task create -t hello-world:{{.Build.ID}} -n hello-world -r MyRegistry -c https://github.com/Azure-Samples/acr-build-helloworld-node.git --commit-trigger-enabled false" is not a valid command.
	the following arguments are required: --git-access-token.


    Help-Entry: `acr build-task show-build` - 
	There is a violation:
	"az acr build-task show-build -n MyBuildTask -r MyRegistry --build-id aab1 -o table" is not a valid command.
	unrecognized arguments: -n MyBuildTask.


    Help-Entry: `acr task update` - 
	There is a violation:
	"az acr task update -n MyTask -r MyRegistry --base-image-trigger All --status Disabled" is not a valid command.
	ambiguous option: --base-image-trigger could match --base-image-trigger-enabled, --base-image-trigger-type.


    Help-Entry: `acs create` - 
	There are 2 violations:
	"az acs create -g MyResourceGroup -n MyContainerService --agent-profiles '["
	has no closing quotation. Tip: to continue an example command on the next line, use a "\" followed by a newline.
	If needed, you can escape the "\", like so "\\"

	"az acs create -g MyResourceGroup -n MyContainerService --agent-profiles '["
	has no closing quotation. Tip: to continue an example command on the next line, use a "\" followed by a newline.
	If needed, you can escape the "\", like so "\\"


    Help-Entry: `aks install-connector` - 
	There is a violation:
	"az aks install-connector --name MyManagedCluster --resource-group MyResourceGroup    --connector-name aci-connector --service-principal <SPN_ID> --client-secret <SPN_SECRET>    --aci-resource-group <ACI resource group>" is not a valid command.
	unrecognized arguments: resource group>.
```

Reply:
Success looks like this:
```
$ azdev cli-lint --rules faulty_help_example_parameters_rule
Initializing linter with command table and help files...

	SKIPPING example: az storage account create -n MyStorageAccount -g MyResourceGroup -l westus --account-type Standard_LRS
	as its max profile is 2017-03-09-profile, instead of latest.
-  pass: faulty_help_example_parameters_rule 
```